Plugin.pm

```diff
@@ -995,6 +995,13 @@ sub QobuzGetTracks {
 		my $noComposer = 0;
 		my $workHeadingPos = 0;
 		my $workPlaylistPos = $prefs->get('workPlaylistPosition');
+		my $currentComposer = "";
+		my $lastComposer = "";
+		my $worksWorkId = "";
+		my $worksWorkIdE = "";
+		my $lastWorksWorkId = "";
+		my $discontigWorks;
+		my $workComposer;
 
 		foreach my $track (@{$album->{tracks}->{items}}) {
```

Comment:
IIRC the works handling is optional, but it seems this code is always executed? Can the processing be made optional, too? I could imagine that this has become pretty heavy on CPU if iterated many times. Maybe you could even move all of the processing into it's own `sub`, to refactor things into manageable and more readable chunks of code?

Reply:
The works handling was there before we started on the classical enhancements road, so it wasn't optional. It's just that it only kicked in when Qobuz had populated work, which we've only ever seen for classical releases.

 The Classical Enhancements project was mainly about deriving sensible works tags when Qobuz didn't provide them. That code is restricted to classical genres only and controlled by the user pref, but the works processing, as such, never was. 

Having said that, while the pre-Classical Enhancements version of the plug-in created work heading playlists when Qobuz provided a work tag, we no longer seem to do so if the preference is switched off, so I'm investigating further.

As to a separate subroutine, I'm not sure I agree - I think the code is easier to read inline, and it's quite well demarcated in the "if works..." block. But maybe we should head that up with a comment explaining what the block is doing.

Reply:
As you like. Was only a suggestion. I personally don't like long chunks of code. Even if a section is only used once, by putting it inside its own function I feel it's cleaner, or at least should be easier to keep things in one place (eg. accidentally overwriting/re-using variables).